cypress/e2e/tests/migration/applicationinventory/analysis/custom-rules.test.ts

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright © 2021 the Konveyor Contributors (https://konveyor.io/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/// <reference types="cypress" />
+
+import {
+    exists,
+    getRandomAnalysisData,
+    getRandomApplicationData,
+    login,
+} from "../../../../../utils/utils";
+import { AnalysisStatuses, CredentialType, UserCredentials } from "../../../../types/constants";
+import * as data from "../../../../../utils/data_utils";
+import { Analysis } from "../../../../models/migration/applicationinventory/analysis";
+import { CredentialsSourceControlUsername } from "../../../../models/administration/credentials/credentialsSourceControlUsername";
+import { Issues } from "../../../../models/migration/dynamic-report/issues/issues";
+import { deleteByList } from "../../../../../utils/utils";
+import { CredentialsMaven } from "../../../../models/administration/credentials/credentialsMaven";
+import { Application } from "../../../../models/migration/applicationinventory/application";
+
+describe(["@tier2"], "Custom Rules in analyses", function () {
+    const applications: Analysis[] = [];
+    let sourceCredential: CredentialsSourceControlUsername;
+    let mavenCredential: CredentialsMaven;
+
+    before("Create test data", function () {
+        login();
+
+        sourceCredential = new CredentialsSourceControlUsername(
+            data.getRandomCredentialsData(
+                CredentialType.sourceControl,
+                UserCredentials.usernamePassword,
+                true
+            )
+        );
+        sourceCredential.create();
+
+        mavenCredential = new CredentialsMaven(
+            data.getRandomCredentialsData(CredentialType.maven, null, true)
+        );
+        mavenCredential.create();
+    });
+
+    beforeEach("Persist session", function () {
+        cy.fixture("application").then(function (appData) {
```

Comment:
Why do we need to load fixtures before each test ? Isn't it enough to load it in before method ?

Reply:
If you're talking about the same logic I applied in this PR: https://github.com/konveyor/tackle-ui-tests/pull/929#discussion_r1462038680

It is not the same case, to access using the `this` keyword, the fixture needs to be loaded before each tests. 

I was using the `beforeEach` approach to aim for consistency, however there are some ways to avoid loading the fixtures in that hook, look at the latest commit, a way is to use the shared lexical scope to store the value, this works and we could get rid of lot of `beforeEach` hooks, the problem is that in this case we'd have to use the `any` type (which I'm not really a fan) because we don't have a class defined for the specific properties of both json files.

Let me know if you prefer this approach and maybe I can work to extend it to other tests and share that with the team

Reply:
Since we cannot access this without loading fixtures in "beforeEach" Let's keep it this way .

Reply:
Done